content/payments/bbps/resources/oauth.mdx

```diff
@@ -7,15 +7,88 @@ visible_in_sidebar: true
 
 ## OAuth 2.0
 
-OAuth 2.0 is an open standard for authorisation and works over HTTPS. In Setu's context, it is used to authorise API requests with access tokens. Read more about OAuth 2.0 <a href="https://tools.ietf.org/html/rfc6749" target="_blank">here ↗</a>.
+OAuth 2.0 is an open standard for authorization and works over HTTPS. In Setu's context, it can be used in two different occassions. 
+ - Merchant can authorize Setu requests for BBPS
+ - Setu can authorize merchant's requests to Setu Reports API
 
-For Setu products, OAuth keys provide a lot of flexibility—you can setup a single set of keys to authorise requests for multiple product configurations, or authorise requests for a single configuration using multiple sets of keys.
+Read more about OAuth 2.0 <a href="https://tools.ietf.org/html/rfc6749" target="_blank">here ↗</a>.
 
 <hr class="primary" />
 
-### Calling Setu-hosted APIs
+### Setu consuming Merchant OAuth Token API
+
+Setu will make a request to the OAuth Token URL using the credentials configured on Bridge to get the `access_token`. 
+This token would then be used to make `/fetch` and `/fetchReceipt` requests to the `BASE URL` configured on Bridge as follows.
+
+Assume `BASE URL` to be `https://base-url.com`
+
+<CodeBlockWithCopy language="shell">
+  {`curl --location --request POST 'https://base-url.com/bills/<fetch/fetchReceipt>' \\
+    --header 'Content-Type: application/json' \\
+    --header 'Authorization: Bearer <access_token>' \\
+    --data '{......}'
+    `}
+</CodeBlockWithCopy>
+
+Refer to the below curl requests and responses based on the chosen `OAuth2.0 scheme` on Bridge
+
+- For `Client credentials`
+
+<CodeBlockWithCopy language="shell">
+  {
+    `
+    curl --location --request POST 'https://<Token URL>' \\
```

Comment:
What is this Token URL? Can we specify it here or link it to another page with the info? 
Same with line 62

Reply:
@santosh-setu

Reply:
@aditya-67 I've made some modifications to this doc. Please take a look n lmk